app/views/questionnaire.py

```diff
@@ -566,12 +566,24 @@ def update_questionnaire_store_with_form_data(questionnaire_store, location, ans
                 _remove_answer_from_questionnaire_store(answer_id, questionnaire_store)
 
             if answer:
+                latest_answer_store_hash = questionnaire_store.answer_store.get_hash()
                 questionnaire_store.answer_store.add_or_update(answer)
+                _remove_dependent_answers_from_completed_blocks(answer_id, questionnaire_store, latest_answer_store_hash)
```

Comment:
i think that `if latest_answer_store_hash != questionnaire_store.answer_store.get_hash():` should wrap this function call.
otherwise it hides that is it not always run

Reply:
I did that originally but it would fail lint as too complex.

Reply:
tut tut

Reply:
when are you going?